### ISSUE TITLE
Hardclips at the start of the read can cause java.lang.StringIndexOutOfBoundsException

### ISSUE THREAD

**Issue by nh13** [AstraZeneca-NGS/VarDictJava#74]
If `n-1 < 0` then the following will cause an `java.lang.StringIndexOutOfBoundsException`.  I think this `if` should have a guard: `0 < n`.  This can occur if the start of the read is hard-clipped.

```
 if (getVariationMaybe(hash, start - 1, ref.get(start - 1)) != null
    && isHasAndEquals(querySequence.charAt(n - 1), ref, start - 1)) {
```

https://github.com/AstraZeneca-NGS/VarDictJava/blob/master/src/main/java/com/astrazeneca/vardict/VarDict.java#L1934

**Comment by nh13** [comment]
This is also likely an issue:
```
char q1 = queryQuality.charAt(n - 1);
```
https://github.com/AstraZeneca-NGS/VarDictJava/blob/master/src/main/java/com/astrazeneca/vardict/VarDict.java#L1979

**Comment by nh13** [comment]
Fixed by #50
